docs/_posts/2017-09-26-react-v16.0.md

```diff
@@ -0,0 +1,209 @@
+  return [
+    // Don't forget the keys :)
+    <li key="A">First item</li>,
+    <li key="B">Second item</li>,
+    <li key="C">Third item</li>,
+  ];
+}
+```
+
+In the future, we'll likely add a special fragment syntax to JSX that doesn't require keys.
+
+We've added support for returning strings, too:
+
+```js
+render() {
+  return 'Look ma, no spans!';
+}
+```
+
+[See the full list of supported return types](/react/docs/react-component.html#render).
+
+### Better error handling
+
+Previously, runtime errors during rendering could put React in a broken state, producing cryptic error messages and requiring a page refresh to recover. To address this problem, React 16 uses a more resilient error-handling strategy. By default, if an error is thrown inside a component's render or lifecycle methods, the whole component tree is unmounted from the root. This prevents the display of corrupted data. However, it's probably not the ideal user experience.
+
+Instead of unmounting the whole app every time there's an error, you can use error boundaries. Error boundaries are special components that capture errors inside their subtree and display a fallback UI in its place. Think of error boundaries like try-catch statements, but for React components.
+
+For more details, check out our [previous post on error handling in React 16](/react/blog/2017/07/26/error-handling-in-react-16.html).
+
+
+### Portals
+
+Portals provide a first-class way to render children into a DOM node that exists outside the DOM hierarchy of the parent component.
+
+```js
+render() {
+  // React does *not* create a new div. It renders the children into `domNode`.
+  // `domNode` is any valid DOM node, regardless of its location in the DOM.
+  return ReactDOM.createPortal(
+    this.props.children,
+    domNode,
+  );
+}
+```
+
+See a full example in the [documentation for portals](/react/docs/portals.html).
+
+### Better server-side rendering
+
+React 16 includes a completely rewritten server renderer. It's really fast. It supports **streaming**, so you can start sending bytes to the client faster. And thanks to a [new packaging strategy](#reduced-file-size) that compiles away `process.env` checks (Believe it or not, reading `process.env` in Node is really slow!), you no longer need to bundle React to get good server-rendering performance.
+
+Core team member Sasha Aickin wrote a [great article describing React 16's SSR improvements](https://medium.com/@aickin/whats-new-with-server-side-rendering-in-react-16-9b0d78585d67). According to Sasha's tests, server rendering in React 16 is roughly **three times faster** than React 15. "When comparing against React 15 with `process.env` compiled out, there’s about a 2.4x improvement in Node 4, about a 3x performance improvement in Node 6, and a full 3.8x improvement in the new Node 8.4 release. And if you compare against React 15 without compilation, React 16 has a full order of magnitude gain in SSR in the latest version of Node!"
```

Comment:
Should we use quote formatting here with >?

Reply:
Maybe worth calling out it's a synthetic bench like the article does.

Reply:
I tried a blockquote but that what we to format those yellow "Note: " boxes :(